_posts/2023-06-11-ayb-helping-you-own-your-data.md

```diff
@@ -0,0 +1,140 @@
+---
+layout: post
+title: 'ayb: Helping you own your data'
+date: 2023-06-11 22:45 +0000
+---
+## Introduction
+
+Today's databases ubiquitous, but frustratingly inaccessible to most people. Your own data likely lives in thousands of databases at various companies and organizations, but if you wanted to create a database for yourself in which to store data and share it, you need the skills of both a system administrator and a software engineer. By virtue of the complexity of database management systems (and market forces), your data either lives in other people's databases or outside of a database in text files and spreadsheets.
+
+Three groups of people in particular feel the pain of databases being hard to set up: students that are just getting started, sharers like journalists and scientist who are experts in fields other than hosting databases, and anyone who wants to have sovereignty over their data but isn't sure how to get started. We'll dive into each of these groups more in [students, sharers, and sovereigns](#students-sharers-and-sovereigns).
+
+I've been working on a project called [ayb](https://github.com/marcua/ayb), which is a relational database management system that makes it easier to to create databases, share them with collaborators, and query them from a web application or the command line. With `ayb`, [all your base can finally belong to you](https://www.youtube.com/watch?v=qItugh-fFgg). What does `ayb` offer?
```

Comment:
```suggestion
To make this concrete, imagine a journalist or student that's looking to create a database around a new dataset and build a visualization they host on a website. Using existing database tools, they have to secure a machine, install Postgres/MySQL, update various configuration settings to allow incoming connections, and create and host a web application (likely managing another machine to accomplish this) to gatekeep the database credentials. Database-as-a-Service providers might reduce the administration overhead in this scenario, but they introduce new complexities like parameter groups and don't eliminate the complexity of the system needed to get up and running. What if our user could: 1) Register for an account on a service provided by their newsroom or school, 2) Issue a `create database` instruction and make their database publicly accessible in read only mode, and 3) Issue SQL from the command line or over HTTP through JavaScript?

Toward that vision, I've been working on a project called [ayb](https://github.com/marcua/ayb), which is a relational database management system that makes it easier to to create databases, share them with collaborators, and query them from a web application or the command line. With `ayb`, [all your base can finally belong to you](https://www.youtube.com/watch?v=qItugh-fFgg). What does `ayb` offer?
```

Reply:
Eugene:
* The key difference/constraint from a static page is that users should be able to update the data in the database.
* Make the distinction: There are two users. One is the user thay sets up ayb as a daas, and thr other is the user that uses the ayb service to deploy an app purely in jabascript/client. Basically it’s “roll your own hosted database service in one command”